File: leetcode/typescript/2635-apply-transform-over-each-element-in-array.ts
const map = <T, U>(arr: T[], fn: (item: T, index: number) => U): U[] => {
  const newArray: U[] = new Array(arr.length); 
  for(let i=0; i<arr.length; ++i) { 
      newArray[i] = fn(arr[i], i); 
  };
  return newArray;
};

File: leetcode/python3/2635_apply_transform_over_each_element_in_array.py
"""
2635. Apply Transform over Each Element in Array 
Problem Link: https://leetcode.com/problems/apply-transform-over-each-element-in-array/
Solution Link (Personal):
"""

from typing import List, Callable

def map(arr: List, fn: Callable) -> List:

    new_arr = [None] * len(arr) 
    for i, item in enumerate(arr): 
        new_arr[i] = fn(item, i) 
    return new_arr 
